### ISSUE TITLE
1.1.1b SSL_ERROR_SYSCALL occurs when SSL_accept is invoked in a thread 

### ISSUE THREAD

**Issue by hufenghuanyu1989** [openssl/openssl#9024]
After I upgraded the version of OpenSSL from 1.0.2h to 1.1.1b, the following problems occurred on my server:
SSL_ERROR_SYSCALL had happend while using SSL_accept

The following is Pseudo code

``` C
int main(){
    SSL_library_init();
    SSL_load_error_strings();
    ct=SSL_CTX_new(SSLv23_method());
    ...
   SSL_CTX_set_session_cache_mode(ct,SSL_SESS_CACHE_OFF)
   socket();
   bind();
   ioctl()
   listen()
   select()
  if(FD_ISSET(fd,&readfd)
  {
     accept()
     ioctl()
     pthread_create(threadID，NULL，fun，fd)
  }
  sleep(200000)
}

fun()
{
    select()
        BIO_new_socket()
         SSL_new()
         SSL_set_bio()
            SSL_accept
            switch(SSL_get_error)
             case SSL_ERROR_NONE
                     success
             case SSL_ERROR_WANT_READ
                     retry
             case SSL_ERROR_WANT_WRITE
                    retry
             case SSL_ERROR_SYSCALL:
                    error
}
```

It is noteworthy that SSL_accept is invoked in the thread,
If you put it(SSL_accept ) in the main process, it will succeed.

**Comment by mattcaswell** [comment]
Does errno tell you anything useful?

**Comment by hufenghuanyu1989** [comment]
> Does errno tell you anything useful?
errno is 0

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev, barring further input
